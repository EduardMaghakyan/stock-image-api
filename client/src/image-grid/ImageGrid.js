import React from "react";

const ImageGrid = ({ images }) => {
  return (
    <div className="image-grid container">
      <div className="row">
        {images.map(imageData => {
          let { id, url, context } = imageData;
          return (
            <div key={id} className="image-grid-item col-md-6">
              <div className="card box-shadow">
                <img
                  className="image-grid-item-image card-img-top"
                  src={url}
                  alt={url}
                />
                <div className="card-body justify-content-center align-items-center">
                  <a
                    href={context}
                    className="btn btn-primary image-grid-item-url"
                    target="_blank"
                  >
                    Go to source
                  </a>
                </div>
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default ImageGrid;
