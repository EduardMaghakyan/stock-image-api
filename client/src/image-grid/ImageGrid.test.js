import React from "react";
import ReactDOM from "react-dom";
import ImageGrid from "./ImageGrid";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });
describe("<ImageGrid />", () => {
  // It should render.
  it("should render empty", () => {
    const wrapper = shallow(<ImageGrid images={[]} />);
    expect(wrapper.find(".image-grid-item").exists()).toBe(false);
  });

  // It should render image-grid-items
  it("should render 1 item", () => {
    const wrapper = shallow(
      <ImageGrid
        images={[
          {
            id: "unique-id",
            url: "https://someurl/image.png",
            context: "https://someurl/contextn.html",
            snippet: "Incredible image"
          }
        ]}
      />
    );
    expect(wrapper.find(".image-grid-item").exists()).toBe(true);
    expect(wrapper.find(".image-grid-item").length).toBe(1);
    expect(wrapper.find(".image-grid-item-image").prop("src")).toBe(
      "https://someurl/image.png"
    );
    expect(wrapper.find(".image-grid-item-url").prop("href")).toBe(
      "https://someurl/contextn.html"
    );
  });

  // Compare with sanpshot
  test("matches saved snapshot", () => {
    const wrapper = shallow(<ImageGrid images={[]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
