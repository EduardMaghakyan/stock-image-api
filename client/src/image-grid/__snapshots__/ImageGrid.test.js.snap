// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageGrid /> matches saved snapshot 1`] = `
<div
  className="image-grid container"
>
  <div
    className="row"
  />
</div>
`;
